gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryAndGatewayDtoConverter.java

```diff
@@ -284,37 +287,57 @@ public static Operator convertOperator(
     throw new IllegalArgumentException("Unsupported operator " + operator.name());
   }
 
-  private static ColumnIdentifier.Builder convertToQueryColumnIdentifier(
+  private static AttributeExpression convertToQueryAttributeExpression(
+      org.hypertrace.gateway.service.v1.common.AttributeExpression attributeExpression) {
+    AttributeExpression.Builder builder =
+        AttributeExpression.newBuilder().setAttributeId(attributeExpression.getAttributeId());
+
+    if (attributeExpression.hasAlias()) {
+      builder.setAlias(attributeExpression.getAlias());
+    }
+    if (attributeExpression.hasSubpath()) {
+      builder.setSubpath(attributeExpression.getSubpath());
+    }
+    return builder.build();
+  }
+
+  private static AttributeExpression convertToQueryAttributeExpression(
       org.hypertrace.gateway.service.v1.common.ColumnIdentifier columnIdentifier) {
-    return ColumnIdentifier.newBuilder()
-        .setColumnName(columnIdentifier.getColumnName())
-        .setAlias(columnIdentifier.getAlias());
+    AttributeExpression.Builder builder =
+        AttributeExpression.newBuilder().setAttributeId(columnIdentifier.getColumnName());
+
+    Optional.of(columnIdentifier.getAlias())
+        .filter(Predicate.not(String::isEmpty))
```

Comment:
Curious Q: Did we observe any issue as prior we were not checking for an empty string? If we are doing this here, do we need to check if not empty in the above function - `convertToQueryAttributeExpression`?

Reply:
On 295, we do this check in a more formal way via `hasAlias()`. This works because on the new GW AttributeExpression message, alias is optional and thus generates the presence check. On the existing column identifier message being used here, because it is not optional the only way to check if it's set is comparing it to the default value (empty string).

Prior to this change, when going from one col identifier to another, the check wasn't needed more or less for the same reason. The QS version of alias didn't support the presence check leaving no difference between an unset alias and one set to an empty string. Now, if we propagated the empty string col identifier alias into attribute expression, `.hasAlias()` would incorrectly return true.

A bit long winded, but does that make sense?

Reply:
Hmm!! Got it.